Doc: github default branch (#29264)
sourcegraph/sourcegraph: doc/admin/external_service/github.md

<div markdown-func=jsonschemadoc jsonschemadoc:path="admin/external_service/github.schema.json">[View page on docs.sourcegraph.com](https://docs.sourcegraph.com/admin/external_service/github) to see rendered content.</div>

## Default branch

> Sourcegraph displays search results from the default branch of a repository when no `revision:` parameter is specified. If you'd like the search results to be displayed from another branch by default, you may [change a repo's default branch on the github repo settings page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch). If this is not an option, consider using [search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) instead. 

Reviewer: Suggest to throw a link to the documentation on the "`revision:`" portion of the first sentence:

https://docs.sourcegraph.com/code_search/reference/queries#repository-revisions

Author: Link added ✅

> GitHub uses the master branch as the default branch of a repository. This means that searches targeting a synched github repo will automatically default to results from the master branch. If however you'd like search results to be displayed from another repository by default you may [change a repo's default branch on the github repo settings page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch). If this is not an option consider using [search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) instead. 

Reviewer 2: ```suggestion
GitHub uses the default branch of a repository for searches and the search results will be shown from the default branch. If you'd like the search results to be displayed from another branch by default, you may [change a repo's default branch on the github repo settings page](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch). If this is not an option, consider using [search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts) instead. 
```

Author: Thanks for the clarifications!
